docs/preview/02-Features/02-message-handling/01-service-bus.md

```diff
@@ -140,111 +131,105 @@ public class Startup
-        services.AddServiceBusTopicMessagePumpUsingManagedIdentity(
-            topicName: properties.EntityPath,
-            subscriptionName: "Receive-All", 
-            fullyQualifiedNamespace: "<your-namespace>.servicebus.windows.net"
-            // The optional client id to authenticate for a user assigned managed identity. More information on user assigned managed identities cam be found here:
-            // https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-a-user-assigned-managed-identity-works-with-an-azure-vm
-            clientId: "<your-client-id>");
+### Bring your own deserialization
+You can also choose to extend the built-in message deserialization with additional deserializer to meet your needs. 
+This allows you to easily deserialize into different message formats or reuse existing (de)serialization capabilities that you already have without altering the message router. 
 
-        services.AddServiceBusTopicMessagePump(
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING",
-            options => 
-            {
-                // Indicate whether or not messages should be automatically marked as completed 
-                // if no exceptions occurred and processing has finished (default: true).
-                options.AutoComplete = true;
+You start by implementing an `IMessageBodySerializer`. The following example shows how an expected type can be transformed to something else. 
+The result type (in this case `OrderBatch`) will be then be used to check if there is an `IAzureServiceBusMessageHandler` registered with that message type.
 
-                // Indicate whether or not the message pump should emit security events (default: false).
-                options.EmitSecurityEvents = true;
+```csharp
+using Arcus.Messaging.Pumps.Abstractions.MessageHandling;
 
-                // The amount of concurrent calls to process messages 
-                // (default: null, leading to the defaults of the Azure Service Bus SDK message handler options).
-                options.MaxConcurrentCalls = 5;
+public class OrderBatchMessageBodySerializer : IMessageBodySerializer
+{
+    public async Task<MessageResult> DeserializeMessageAsync(string messageBody)
+    {
+        var serializer = new XmlSerializer(typeof(Order[]));
+        using (var contents = new MemoryStream(Encoding.UTF8.GetBytes(messageBody)))
+        {
+            var orders = (Order[]) serializer.Deserialize(contents);
+            return MessageResult.Success(new OrderBatch(orders));
+        }
+    }
+}
+```
 
-                // The unique identifier for this background job to distinguish 
-                // this job instance in a multi-instance deployment (default: guid).
-                options.JobId = Guid.NewGuid().ToString();
+The registration of these message body handlers can be done just as easily as an `IMessageSerializer`:
```

Comment:
This sentence is confusing.  We just showed an example of a custom serializer, but this sentence says that registering a body handler (message handler I suppose ? ) is as simple as registering a serializer.

I think you want to say here that registering a serializer is as easy as registering a message-handler ?

Reply:
Ah, yes, 'as easily as a message handler' is correct. Thx.

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/268/commits/cd44eb2753ac33d70bef057cea45e54bbf30fb40